tests/rptest/tests/group_membership_test.py

```diff
@@ -113,3 +116,187 @@ def _check_no_duplicates():
+        rpk = RpkTool(self.redpanda)
+
+        topic = "test_topic"
+        spec = TopicSpec(name=topic, partition_count=1)
+        self.redpanda.create_topic(spec)
+        self._wait_topic_is_ready(topic)
+
+        for i in range(100):
+            payload = str(random.randint(0, 1000))
+            offset = rpk.produce(topic, "", payload, timeout=5)
+
+        group_1 = "g1"
+        metric_key = (topic, "0")
+        for i in range(10):
+            rpk.consume(topic, group=group_1, n=10)
+            metrics_offsets = self._get_offset_from_metrics(group_1)
+            assert metric_key in metrics_offsets
+            assert metrics_offsets[metric_key] == (i + 1) * 10
+
+        group_2 = "g2"
+        rpk.consume(topic, group=group_2, n=50)
+        gr_2_metrics_offsets = self._get_offset_from_metrics(group_2)
+        assert metric_key in gr_2_metrics_offsets
+        assert gr_2_metrics_offsets[metric_key] == 50
+
+        rpk.group_seek(group_1, to_group=group_2)
+        gr_1_metrics_offsets = self._get_offset_from_metrics(group_1)
+        assert metric_key in gr_1_metrics_offsets
+        assert gr_1_metrics_offsets[metric_key] == 50
+
+        rpk.group_seek(group_2, to="start")
+        gr_2_metrics_offsets = self._get_offset_from_metrics(group_2)
+        assert metric_key in gr_2_metrics_offsets
+        assert gr_2_metrics_offsets[metric_key] == 0
+
+        self.redpanda.delete_topic(topic)
+        metrics_offsets = self._get_offset_from_metrics(group_1)
+        assert metrics_offsets is None
+
+    @cluster(num_nodes=3)
+    def test_multiple_topics_and_partitions(self):
+        rpk = RpkTool(self.redpanda)
+
+        topic_1 = "test_topic_1"
+        spec = TopicSpec(name=topic_1, partition_count=3, replication_factor=3)
+        self.redpanda.create_topic(spec)
+        self._wait_topic_is_ready(topic_1)
+
+        topic_2 = "test_topic_2"
+        spec = TopicSpec(name=topic_2, partition_count=3, replication_factor=3)
+        self.redpanda.create_topic(spec)
+        self._wait_topic_is_ready(topic_2)
+
+        for i in range(100):
+            key = str(random.randint(0, 1000))
+            payload = str(random.randint(0, 1000))
+            rpk.produce(topic_1, key, payload, timeout=5)
+            rpk.produce(topic_2, key, payload, timeout=5)
+
+        group = "group"
+        rpk.consume(topic_1, group=group, n=100)
+        rpk.consume(topic_2, group=group, n=100)
+
+        metrics_offsets = self._get_offset_from_metrics(group)
+
+        self.redpanda.logger.debug(f"KEK: {metrics_offsets}")
+        sum_1 = 0
+        sum_2 = 0
+        assert len(metrics_offsets) == 6
+        for i in range(3):
+            assert (topic_1, str(i)) in metrics_offsets
+            assert metrics_offsets[(topic_1, str(i))] > 0
+            sum_1 += metrics_offsets[(topic_1, str(i))]
+
+            assert (topic_2, str(i)) in metrics_offsets
+            assert metrics_offsets[(topic_2, str(i))] > 0
```

Comment:
why are we not able to assert a precise value?

by the way, you might consider using `rpk` or the `kafka-consumer-groups.sh` tool to set offset data directly allowing you to create precise equality tests on offsets, rather than depending on the indirect behavior of consumers to update the offsets in the consumer. depending on that indirect behavior may lead to tests that aren't robust.